File: CS/WpfApplication1/GridControlSpellChecker.cs
using DevExpress.Xpf.Editors;
using DevExpress.Xpf.Grid;
using DevExpress.Xpf.SpellChecker;
using System;
using System.Windows;
using System.Windows.Threading;

namespace WpfApplication1
{
    public class GridControlSpellChecker : DXSpellCheckerBase<GridControl>
    {
        GridControl Grid { get { return AssociatedObject; } }
        protected override void OnAttached()
        {
            base.OnAttached();
            Grid.Dispatcher.BeginInvoke(new Action(() => SubscribeToEvents()), DispatcherPriority.Loaded);
        }
                   
        #region #SubscribeToEvents
        private void SubscribeToEvents()
        {
            SpellChecker.CheckCompleteFormShowing += Checker_CheckCompleteFormShowing;
            CardView cardView = Grid.View as CardView;
            if (cardView != null)
                cardView.ShownEditor += CardView_ShownEditor;
        }
        private void UnsubscribeFromEvents()
        {
            SpellChecker.CheckCompleteFormShowing -= Checker_CheckCompleteFormShowing;
            CardView cardView = Grid.View as CardView;
            if (cardView != null)
                cardView.ShownEditor -= CardView_ShownEditor;
        }
        protected override void OnDetaching()
        {
            UnsubscribeFromEvents();
            base.OnDetaching();
        }
        #endregion #SubscribeToEvents

        #region #ShownEditor
        private void CardView_ShownEditor(object sender, EditorEventArgs e)
        {
            var cardView = (sender as CardView);
            BaseEdit activeEditor = cardView.ActiveEditor;
            if (SpellChecker.SpellCheckMode == DevExpress.XtraSpellChecker.SpellCheckMode.OnDemand)
                CheckActiveEditor(activeEditor);
        }
        void CheckActiveEditor(BaseEdit activeEditor)
        {
            activeEditor.Dispatcher.BeginInvoke(new Action(() =>
            {
                if (SpellChecker.CanCheck(activeEditor))
                    SpellChecker.Check(activeEditor);
            }), DispatcherPriority.Loaded);
        }
        #endregion #ShownEditor

        #region #CompleteFormShowing
        void Checker_CheckCompleteFormShowing(object sender, DevExpress.XtraSpellChecker.FormShowingEventArgs e)
        {
            e.Handled = true;
        }
        #endregion #CompleteFormShowing
    }
}

File: VB/WpfApplication1/GridControlSpellChecker.vb
Imports DevExpress.Xpf.Editors
Imports DevExpress.Xpf.Grid
Imports DevExpress.Xpf.SpellChecker
Imports System
Imports System.Windows
Imports System.Windows.Threading

Namespace WpfApplication1
	Public Class GridControlSpellChecker
		Inherits DXSpellCheckerBase(Of GridControl)

		Private ReadOnly Property Grid() As GridControl
			Get
				Return AssociatedObject
			End Get
		End Property
		Protected Overrides Sub OnAttached()
			MyBase.OnAttached()
			Grid.Dispatcher.BeginInvoke(New Action(Sub() SubscribeToEvents()), DispatcherPriority.Loaded)
		End Sub

		#Region "#SubscribeToEvents"
		Private Sub SubscribeToEvents()
			AddHandler SpellChecker.CheckCompleteFormShowing, AddressOf Checker_CheckCompleteFormShowing
			Dim cardView As CardView = TryCast(Grid.View, CardView)
			If cardView IsNot Nothing Then
				AddHandler cardView.ShownEditor, AddressOf CardView_ShownEditor
			End If
		End Sub
		Private Sub UnsubscribeFromEvents()
			RemoveHandler SpellChecker.CheckCompleteFormShowing, AddressOf Checker_CheckCompleteFormShowing
			Dim cardView As CardView = TryCast(Grid.View, CardView)
			If cardView IsNot Nothing Then
				RemoveHandler cardView.ShownEditor, AddressOf CardView_ShownEditor
			End If
		End Sub
		Protected Overrides Sub OnDetaching()
			UnsubscribeFromEvents()
			MyBase.OnDetaching()
		End Sub
		#End Region ' #SubscribeToEvents

		#Region "#ShownEditor"
		Private Sub CardView_ShownEditor(ByVal sender As Object, ByVal e As EditorEventArgs)
			Dim cardView As CardView = (TryCast(sender, CardView))
			Dim activeEditor As BaseEdit = cardView.ActiveEditor
			If SpellChecker.SpellCheckMode = DevExpress.XtraSpellChecker.SpellCheckMode.OnDemand Then
				CheckActiveEditor(activeEditor)
			End If
		End Sub
		Private Sub CheckActiveEditor(ByVal activeEditor As BaseEdit)
			activeEditor.Dispatcher.BeginInvoke(New Action(Sub()
				If SpellChecker.CanCheck(activeEditor) Then
					SpellChecker.Check(activeEditor)
				End If
			End Sub), DispatcherPriority.Loaded)
		End Sub
		#End Region ' #ShownEditor

		#Region "#CompleteFormShowing"
		Private Sub Checker_CheckCompleteFormShowing(ByVal sender As Object, ByVal e As DevExpress.XtraSpellChecker.FormShowingEventArgs)
			e.Handled = True
		End Sub
		#End Region ' #CompleteFormShowing
	End Class
End Namespace